### PR TITLE
Fixes #9999: Use hash from signature in sharedfiles methods

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9999

### CODE REVIEW

File: tree/30_generic_methods/sharedfile_to_node.cf [hunk 1]
```diff
@@ -69,30 +67,28 @@ bundle agent sharedfile_to_node(target_uuid, file_id, file_path, ttl)
...
       "result"                    usebundle => _classes_combine_ifthenelse("${class_prefix}_exist", "success", "${class_prefix}_uploaded", "${class_prefix}");
 
       "report"                    usebundle => _log("Sharing ${file_path} with ${target_uuid} under the name ${file_id} for ${ttl}", "${old_class_prefix}", "${class_prefix}", @{args});
 
   commands:
-    _stdlib_path_exists_curl.sha512sum::
+    _stdlib_path_exists_curl::
       "/opt/rudder/bin/rudder-sign"
-        args => "${file_path}",
+        args => "${file_path} 1.1",
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_to_node.cf]: This means we require a 4.1 agent.

Comment (peckpeck) [inline tree/30_generic_methods/sharedfile_to_node.cf]: Indeed, we cannot share a file on an older agent

**Comment by peckpeck** [comment]
Commit modified

File: tree/30_generic_methods/sharedfile_from_node.cf [hunk 2]
```diff
@@ -25,6 +25,8 @@
 # ```
 # sharedfile_from_node("2bf1afdc-6725-4d3d-96b8-9128d09d353c", "application.properties", "/opt/application/etc/application.properties")
 # ```
+#  Please not that this methods must be used on a Rudder agent that is connected to
+#  Rudder relay server (>=4.1)
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_from_node.cf]: relay or server

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**